batch/batch.go

```diff
@@ -154,6 +171,15 @@ func (f *Func) Invoke(ctx context.Context, arg interface{}) (interface{}, error)
 
 		// Publish the batchGroup.
 		bctx.pendingBatchGroups[fs] = bg
+	} else {
+		// This is a subsequent invocation of the batch function, so
+		// reset the intervalTimer.
+		if bg.intervalTimer.Stop() {
+			// Only reset the timer if Stop successfully
+			// stopped the timer. Otherwise, the receive channel
+			// will be populated and we'll select on it below.
```

Comment:
Yes, if we fail to stop the timer, that means we've hit a race condition where we requested to stop the timer but the receive channel has already been written to. In this case, we should allow the select loop to read the value and execute the batching function.